src/lib.rs

```diff
@@ -21,6 +22,66 @@ pub struct StartLspServerParams {
     pub options: Option<Value>,
 }
 
+/// The id of a started Language Server.  
+/// This is used to reference the server in future requests.
+pub type LspId = u64;
+
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct StartLspServerResult {
+    pub id: LspId,
+}
+
+pub struct SendLspNotification {}
+
+impl Request for SendLspNotification {
```

Comment:
I'm wondering if we need a response for this. If we want to see if lsp server is dead, we could have a notification from lapce.

Reply:
The idea behind having a response would be that you can handle the error of the spawned LSP not existing. I don't think there's a nice way to do that if `SendLspNotification` was a Notification?  
While we can (and perhaps should) have a notification for if the LSP dies, it might be more annoying to handle the case of your notification not getting anywhere.   
  
My initial reasoning was like:  
If there was some state the plugin is keeping track of that it needs to tell the language server about, like if the plugin registered a panel with an input box, then it may want to know if it failed to inform the language server due to the language server being dead. If the LspExited notification comes in, it doesn't know what the timing was with respect to the notification it sent out... but it can really just store it and resend it later once it restarts the LSP server (if the plugin wants to).
So now that I think about it, I think I agree with you. Most of the ways I'm imagining this having issues are fine with receiving LspExited and `SendLspNotification` being a notification as well.
I'll rework the send lsp logic.